### ISSUE TITLE
On ROCm, MIOpen can't find invoker for convolution forward when using pmap with 8 devices, but works fine with 1 device

### ISSUE THREAD

**Issue by SarthakYadav** [google/jax#14582]
### Description

Hi!

I'm working with AMD GPUs in a cluster environment.

When using pmap with >1 GPUs MIOpen can't seem to find the correct invoker for the convolution forward and raises the following error:

```
MIOpen Error: /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/MLOpen/src/ocl/convolutionocl.cpp:658: No invoker was registered for convolution forward. Was find executed?
```

But it runs just fine with a single GPU. This can also be reproduced with the following code snippet.

```python
from flax import jax_utils
import jax
import flax.linen as nn
import jax.numpy as jnp
rng = jax.random.PRNGKey(0)
conv = nn.Conv(1, kernel_size=(7,7), strides=(2,2), padding="VALID")
var = conv.init(rng, jnp.ones((1, 14, 14, 1)))
x = jax.random.normal(rng, (16, 14, 14, 1))
def forward(batch):
    return conv.apply(var, batch)
### works
o = forward(x).shape

pforward = jax.pmap(forward, axis_name="batch")
x = jax.random.normal(rng, (8, 16, 14, 14, 1))

### doesn't work
print(pforward(x).shape)
```

I've attached full log files from test runs with 1 and 8 devices, 
[logfile_1gpu.txt](https://github.com/google/jax/files/10778977/logfile_1gpu.txt)
[logfile_8gpus.txt](https://github.com/google/jax/files/10778978/logfile_8gpus.txt)

which were collected as follows 
```bash
export MIOPEN_ENABLE_LOGGING=1
export MIOPEN_LOG_LEVEL=5
export MIOPEN_FIND_MODE=3

python jax_simple.py 2>&1 | tee logfile_8gpus.txt
CUDA_VISIBLE_DEVICES=0 python test.py 2>&1 | tee logfile_1gpu.txt
```

If it helps, official `jax.pmap` examples run just fine. Even flax mnist and imagenet examples train just fine, as long as 1 gpu is used. 

Thanks!

### What jax/jaxlib version are you using?

jax v0.4.3, jaxlib v0.4.3

### Which accelerator(s) are you using?

GPU (AMD MI250x)

### Additional system info

Python 3.9, slurm environment, ROCm 5.3.3

**Comment by hawkinsp** [comment]
@rahulbatra85 maybe?

**Comment by rahulbatra85** [comment]
@SarthakYadav  This is not a JAX issue, but coming from XLA on ROCm. 

Did you build Jax/Jaxlib on ROCm yourself?

**Comment by SarthakYadav** [comment]
@rahulbatra85 Yes, jaxlib was built by the cluster admins.

**Comment by rahulbatra85** [comment]
Sorry for the delayed response. As a workaround can you try running with this environment variable `TF_ROCM_USE_IMMEDIATE_MODE=1 `
Note, this may affect performance.

I will update once I have a fix for this.

**Comment by brett-koonce** [comment]
See also #18906.

**Comment by rahulbatra85** [comment]
@brett-koonce This should be fixed in latest release 0.4.25 https://github.com/ROCm/jax/releases/tag/jaxlib-v0.4.25.
Shouldn't  need to set environment variable manually.

**Comment by SarthakYadav** [comment]
> @brett-koonce This should be fixed in latest release 0.4.25 https://github.com/ROCm/jax/releases/tag/jaxlib-v0.4.25. Shouldn't need to set environment variable manually.

Yes, I can attest to that. Since creating this issue, I've successfully done quite a few experiments with more recent versions of jaxlib without any problems. Closing this.
